input-profile.ts

```diff
@@ -0,0 +1,322 @@
+            for (let i = children.length - 1; i >= 0; --i) {
+                stack.push(children[i]);
+            }
+        }
+        return components;
+    }
+
+    setComponentsActive(active: boolean) {
+        const comps = this.defaultControllerComponents;
+        if (!comps) return;
+        for (let i = 0; i < comps.length; ++i) {
+            comps[i].active = active;
+        }
+    }
+    // Save data to cache
+    saveToCache(key, data) {
+        try {
+            // Convert data to JSON string before storing
+            const jsonString = JSON.stringify(data);
+
+            // Save to local storage
+            localStorage.setItem(key, jsonString);
+        } catch (error) {
+            console.error('Error saving to cache:', error);
+        }
+    }
+
+    // Retrieve data from cache
+    retrieveFromCache(key) {
+        try {
+            // Retrieve data from local storage
+            const jsonString = localStorage.getItem(key);
+
+            // Parse JSON string to get the original data
+            const data = JSON.parse(jsonString);
+
+            return data;
+        } catch (error) {
+            console.error('Error retrieving from cache:', error);
+            return null;
+        }
+    }
+
+    onInputSourcesChange(event: any) {
+        if (this.modelLoaded && !this.mapToDefaultController) {
+            this.setComponentsActive(false);
+        }
+
+        event.added.forEach((xrInputSource: any) => {
+            if (xrInputSource.hand != null) return;
+            if (this.handedness != xrInputSource.handedness) return;
+            var profile = this.path + xrInputSource.profiles[0];
+            if (this.customProfileFolder !== '') profile = this.customProfileFolder;
+            this.url = profile + '/profile.json';
+            this.ProfileJSON = this.retrieveFromCache(this.url);
+            if (this.ProfileJSON != null) {
+                console.log('Loaded Profile From Cache ');
+                this.loadAndMapGamepad(profile, xrInputSource);
+            } else {
+                fetch(this.url)
+                    .then((res) => res.json())
+                    .then((out) => {
+                        this.ProfileJSON = out;
+                        console.log('Profile downloaded and loaded from ' + this.url);
+                        this.saveToCache(this.url, out);
+                        this.loadAndMapGamepad(profile, xrInputSource);
+                    })
+                    .catch((e) => {
+                        console.error('failed to load profile from' + this.url);
+                        console.error(e);
+                    });
+            }
+        });
+    }
+
+    isModelLoaded() {
+        return this.controllerModel !== null;
+    }
+
+    loadAndMapGamepad(profile: string, xrInputSource: any) {
+        this.gamepad = xrInputSource.gamepad;
+
+        const assetPath = profile + '/' + this.handedness + '.glb';
+        if (this.modelLoaded) return;
```

Comment:
Why this condition? If you don't want to overcall this method, simply don't call it. No need to implicitely prevent a load